flink-state-backends/flink-statebackend-rocksdb/src/test/java/org/apache/flink/state/rocksdb/RocksDBWriteBatchWrapperTest.java

```diff
@@ -96,6 +101,11 @@ public void testAsyncCancellation() throws Exception {
                 // make sure that cancellation is triggered earlier than periodic flush
                 // but allow some delay of cancellation propagation
                 assertThat(i).isLessThan(cancellationCheckInterval * 2);
+                if (i == 0) {
+                    // make sure the registry is closed at least after the first run, so that we
+                    // can verify the cancellation check is validating correctly.
+                    cancellationRequestedFuture.join();
+                }
```

Comment:
In the original comment, it said: `allow some delay of cancellation propagation`. It's the reason why this test is flaky, because we can't assure how much time it'll take for the `cancellation propagation`. With the `cancellationRequestedFuture.join();` here, we can make sure the `cancellation propagation` is done at least after the first run, to make the test reliable.